src/modules/commander/Commander.cpp

```diff
@@ -2000,7 +2007,7 @@ Commander::set_in_air_home_position()
 			ref_pos.project(home.lat, home.lon, home_x, home_y);
 
 			const float home_z = -(home.alt - lpos.ref_alt);
-			fillLocalHomePos(home, home_x, home_y, home_z, NAN);
+			fillLocalHomePos(home, home_x, home_y, home_z, NAN, NAN, NAN);
```

Comment:
When is  `set_in_air_home_position()` called? Is this triggered by calling the MAVLink command?

My concern here is that it is being set to NAN, so it might be swamping the value from the MAVLink command. 

Further, I assume that if you don't have local position you don't have yaw, pitch or roll either.

Reply:
Good question. I didn't check when this is called, I just kept the behavior the same.

Now that I checked, I'm a bit confused what this is for, it depends on the param `COM_HOME_IN_AIR`
https://github.com/PX4/PX4-Autopilot/blob/156b68b252d43dba9728acfa608d8dbfa8d4d12b/src/modules/commander/commander_params.c#L174-L178

It basically allows you to take off and then you determine home to be there where you first get GPS, so I guess if you come out of a building, then home would be a few meters outside your building, when you acquired GPS.